cmd/zed/manage/lakemanage/lakemanage.go

```diff
@@ -192,20 +197,44 @@ func (t *thread) run() {
 		timer := time.NewTimer(0)
 		<-timer.C
 		for t.ctx.Err() == nil {
-			next, err := t.task.run(t.ctx)
+			head, ok := t.head()
+			if !ok {
+				return
+			}
+			next, err := t.task.run(t.ctx, head)
 			if err != nil {
 				t.task.logger().Error("thread exited with error", zap.Error(err))
 				return
 			}
 			if next == nil {
+				// This means there's no further work, but before exiting check
+				// to see if there are any new commits since the task was run.
+				current, ok := t.head()
+				if !ok {
+					return
+				}
+				if current != head {
+					continue
+				}
 				t.task.logger().Info("thread exiting")
 				return
```

Comment:
Nit: Save head across loop iterations to avoid batch to back t.branch.head calls.
```go
	 	var head ksuid.KSUID
		for t.ctc.Err() == nil {
			current, err := t.branch.head(t.ctx)
			if err != nil {
				t.task.logger().Error("error fetching branch head", zap.Error(err))
				return
			}
			if current == head {
				t.task.logger().Info("thread exiting")
				return
			}
			head = current
			next, err := t.task.run(t.ctx, head)
			if err != nil {
				t.task.logger().Error("thread exited with error", zap.Error(err))
				return
			}
			if next == nil {
				// This means there's no further work, but before exiting check
				// to see if there are any new commits since the task was run.
				continue

Reply:
Hell yeah!